### ISSUE TITLE
Data Factory version not detected with template

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule.Rules.Azure#498]
**Description of the issue**

When using a template containing a Data Factory v2 instance, `Azure.DataFactory.Version` incorrectly recommends upgrading to Data Factory V2 for a V2 instance. This is because the `type` property is used instead of `ResourceType`.

**Expected behaviour**

Update rule to use target type which is bound to the correct property depending on if the resource has been exported from Azure/ template.

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **0.16.0-B2009011**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.0.3
PSEdition                      Core
GitCommitId                    7.0.3
OS                             Microsoft Windows 10.0.19041
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
